### PR TITLE
Allow instance edit with a custom iso attached

### PR DESCRIPTION
**Pull request by edlerd**
## Done

- fixed instance edit with attached custom iso

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - create an instance with an attached iso volume
    - edit the instance and hit save, it should succeed

### CODE REVIEW

File: src/util/formDevices.tsx [hunk 1]
```diff
@@ -75,16 +75,24 @@ export const parseDevices = (devices: LxdDevices): FormDevice[] => {
...
+        name: key,
+        bare: item,
+        type: "custom-nic",
+      };
+    }
+
+    if (key === "iso-volume") {
+      return {
+        name: key,
+        bare: item as never,
```
Comment (lorumic) [inline src/util/formDevices.tsx:89]: `item` seems to have a value here, shouldn't this be treated as `unknown`?
```suggestion
        bare: item as unknown,
```

**Review by lorumic — APPROVED** [review]
LGTM, just one small type question.

File: src/util/formDevices.tsx (hunk 1 above)
Comment (edlerd) [reply src/util/formDevices.tsx]: Right, `unknown` is the better choice here. I also had to change the `UnknownDevice` type for this to work though.

**Review by aaryanporwal — APPROVED** [review]
lgtm

**Pull request merged**